plugin/src/main/java/org/opensearch/ml/processor/ModelExecutor.java

```diff
@@ -0,0 +1,218 @@
+                                        tensorArray.add(parseGetDataInTensor(tensor));
+                                    }
+                                } catch (Exception e) {
+                                    // Handle the exception accordingly
+                                    throw new RuntimeException("Error accessing tensor data: " + e.getMessage());
+                                }
+                            }
+                        }
+                        modelOutputValue = tensorArray;
+                    }
+
+                } else {
+                    throw new RuntimeException("Output tensors are null or empty.");
+                }
+            } else {
+                throw new RuntimeException("Model outputs are null or empty.");
+            }
+        } catch (Exception e) {
+            throw new RuntimeException("An unexpected error occurred: " + e.getMessage());
+        }
+        return modelOutputValue;
+    }
+
+    static Object parseGetDataInTensor(ModelTensor tensor) {
+        Object modelOutputValue;
+        if (tensor.getDataType().isInteger()) {
+            modelOutputValue = Arrays.stream(tensor.getData()).map(Number::intValue).map(Integer::new).collect(Collectors.toList());
+        } else if (tensor.getDataType().isFloating()) {
+            modelOutputValue = Arrays.stream(tensor.getData()).map(Number::floatValue).map(Float::new).collect(Collectors.toList());
+        } else if (tensor.getDataType().isString()) {
+            modelOutputValue = Arrays.stream(tensor.getData()).map(String::valueOf).map(String::new).collect(Collectors.toList());
+        } else if (tensor.getDataType().isBoolean()) {
+            modelOutputValue = Arrays
+                .stream(tensor.getData())
+                .map(num -> num.intValue() != 0)
+                .map(Boolean::new)
+                .collect(Collectors.toList());
+        } else {
+            throw new RuntimeException("unsupported data type in prediction data.");
+        }
+        return modelOutputValue;
+    }
+
+    /**
+     * filter model outputs by field name,
+     * supported using dot path look up for objects
+     * when a field name not provided, default to get all prediction outputs
+     */
+    default Object getModelOutputField(Map<String, ?> modelTensorOutputMap, String fieldName, boolean ignoreMissing) throws IOException {
+
+        if (fieldName == null) {
+            return modelTensorOutputMap;
+        } else if (modelTensorOutputMap.containsKey(fieldName) && modelTensorOutputMap != null) {
+            Object filteredOutput = modelTensorOutputMap.get(fieldName);
+            return filteredOutput;
+        } else {
+            try {
+                Object filteredOutput = JsonPath.using(suppressExceptionConfiguration).parse(modelTensorOutputMap).read(fieldName);
+                return filteredOutput;
+            } catch (Exception e) {
+                if (ignoreMissing) {
```

Comment:
What does ignoreMissing mean? Why do we always return `modelTensorOutputMap` for any exception if `ignoreMissing == true` ?

Reply:
ignoreMissing means when a document is missing the field, when document is missing the field, we skip inferencing  on this field. 
or the prediction outcome is missing the field, we ignore filtering the prediction outcomes, and return the whole modelTensorOutputMap.